test/boost/virtual_table_mutation_source_test.cc

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2021 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "db/virtual_table.hh"
+#include "db/system_keyspace.hh"
+#include "schema.hh"
+
+#include "test/lib/mutation_source_test.hh"
+
+#include <seastar/testing/thread_test_case.hh>
+
+class memtable_filling_test_vt : public db::memtable_filling_virtual_table {
+    std::vector<mutation> _mutations;
+public:
+    memtable_filling_test_vt(schema_ptr s, std::vector<mutation> mutations)
+            : memtable_filling_virtual_table(s)
+            , _mutations(std::move(mutations)) {}
+
+    future<> execute(memtable& mt) override {
+        return do_for_each(_mutations, [&mt] (const mutation& m) { mt.apply(m); });
+    }
+};
+
+static void run_tests(populate_fn_ex populate) {
```

Comment:
The downside of doing it like this is that we will surely forget to add calls to new test that we create in the future.

Maybe add a param to run_mutation_source_tests() which allows us to disable f_f_t() tests.

Reply:
Even better, use `make_forwardable()` to make your unforwardable reader forwardable, unless you are talking about the other `f_f_t()`, (hard to tell).

Reply:
@tgrabiec Done.
@denesb Is it a good idea though? Isn't it that the generic fast forwarding wrapper would just iterate over the stream (?). I'm no expert, I just see  that Tomasz disabled it explicitly.

Reply:
The idea is to only use it in tests, where performance is not of concern.

Reply:
Ah, ok, I can do it.

Reply:
@jul-stas It was not really disabled, just not implemented (because it's hard). make_forwardable() makes sense.